api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -141,44 +147,87 @@ public Cohort(String commaSeparatedIds) {
 	 */
 	public String getCommaSeparatedPatientIds() {
 		StringBuilder sb = new StringBuilder();
-		for (Iterator<Integer> i = getMemberIds().iterator(); i.hasNext();) {
-			sb.append(i.next());
-			if (i.hasNext()) {
-				sb.append(",");
-			}
+        for (CohortMembership member : getMembers()) {
+			sb.append(member.getPatient().getPatientId());
+			sb.append(",");
 		}
 		return sb.toString();
 	}
-	
-	public boolean contains(Patient patient) {
-		return getMemberIds() != null && getMemberIds().contains(patient.getPatientId());
-	}
-	
+
+	// TODO: Compare patient objects instead of IDs after removal of deprecated methods
 	public boolean contains(Integer patientId) {
-		return getMemberIds() != null && getMemberIds().contains(patientId);
+		return getMembers() != null && getMembers().stream()
+				.anyMatch(m -> m.getPatient().getPatientId().equals(patientId) &&
+				m.getStartDate().before(new Date()) &&
+				m.getEndDate() == null);
 	}
-	
+
+	public boolean contains(CohortMembership cohortMembership) {
+		return contains(cohortMembership.getPatient().getPatientId());
```

Comment:
I think the semantics of this method need to be clarified in the javadocs, my assumption is that it need to take into consideration the startDate and endDates. Startdate isn't expected to be null.
@bmamlin, @djazayeri do we really need this method? In case scenario would one call it? The useful methods I imagine devs would call are contains(Patient) or contains(Integer)